anndata/core/anndata.py

```diff
@@ -515,7 +507,7 @@ def _init_as_actual(
         self._clean_up_old_format(uns)
 
         # layers
-        self._layers = Layers(self, layers)
+        self._layers = Layers(self, vals=(layers or {}))
```

Comment:
If the case you're trying to deal with is `layers = None`, use an explicit check since other values can coerce to false.

e.g.

```python
self._layers = Layers(self, vals=(layers if layers is not None else {}))
```

Reply:
That’s deliberate. We don’t care if one empty dict or another is used. As long as we don’t mutate them, the effect is the same.

Reply:
answered above

Reply:
My thought was that it should throw an error if it's a type that shouldn't be here, like if it was `False`.

But I had forgotten there's already a solution to this. This should be `self._layers = Layers(self, vals=convert_to_dict(layers))` like the others (above):

```python
self._obsm = AxisArrays(self, 0, vals=convert_to_dict(obsm))
self._varm = AxisArrays(self, 1, vals=convert_to_dict(varm))

self._obsp = PairwiseArrays(self, 0, vals=convert_to_dict(obsp))
self._varp = PairwiseArrays(self, 1, vals=convert_to_dict(varp))
```

Reply:
OK!